This patch file is from Qualcomm Innovation Center, Inc. and is provided under the following license:
Copyright (c) 2025 Qualcomm Innovation Center, Inc. All rights reserved.
SPDX-License-Identifier: BSD-3-Clause-Clear

From ea736d3ba6f7ab86b3a1572a87c5962a575b8b0f Mon Sep 17 00:00:00 2001
From: Devanshi Bansal <quic_devanshi@quicinc.com>
Date: Mon, 13 Jan 2025 14:49:27 +0530
Subject: [PATCH] gl-renderer: choose correct texture target for multiplanar
 format

Upstream-Status: Inappropriate [Downstream]
Signed-off-by: Devanshi Bansal <quic_devanshi@quicinc.com>
---
 libweston/renderer-gl/gl-renderer.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libweston/renderer-gl/gl-renderer.c b/libweston/renderer-gl/gl-renderer.c
index faf3bdf..134638f 100644
--- a/libweston/renderer-gl/gl-renderer.c
+++ b/libweston/renderer-gl/gl-renderer.c
@@ -2840,6 +2840,10 @@ choose_texture_target(struct gl_renderer *gr,
 		}
 	}
 
+
+        if (attributes->n_planes > 1)
+                return GL_TEXTURE_EXTERNAL_OES;
+
 	switch (attributes->format & ~DRM_FORMAT_BIG_ENDIAN) {
 	case DRM_FORMAT_YUYV:
 	case DRM_FORMAT_YVYU:
-- 
2.34.1

